modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/MuleSdkExtensionModelLoader.java

```diff
@@ -28,6 +29,10 @@ public String getId() {
 
   @Override
   protected ExtensionModelParserFactory getExtensionModelParserFactory(ExtensionLoadingContext context) {
-    return new MuleSdkExtensionModelParserFactory();
+    if (MuleSdkExtensionExtensionModelParserFactory.handles(context)) {
```

Comment:
I don't think we should solve this with an if but with inheritance

Reply:
I don't think I understand exactly what you mean.
I'm actually using inheritance (together with polymorphism) to deal with the Application vs Extension cases, however, at some point I need to figure out which concrete class to instantiate. This is were I'm making the decision.
I'm using this class method (`handles`) to figure out which concrete class is capable of handling the current loading request before instantiating it.

Another option I've been contemplating would be to actually use different ExtensionModelLoaders, one for the application and one for the extension. In that case there would be no need for this if.

Reply:
what you said in the last paragraph is exactly what I meant. Use different loaders with a common parent class

Reply:
Done. We have two separate loaders now, each with its ID. The `handles` method was no longer necessary, so I removed it. Each parser is now a little bit simpler I think. Thanks for the suggestion.